api/app/serializers/text_partial_serializer.rb

```diff
@@ -12,6 +13,10 @@ def start_text_section_id
     object.start_text_section_id ||= object.spine[0]
   end
 
+  def sections_map
+    object.spine.map { |id| Hash[id: id.to_s, name: object.text_sections.find(id).name] }
```

Comment:
This should be more forgiving. If, for example, a text section was deleted from a text, any call to this method would trigger a record not found exception due to `text_sctions.find`. Assuming spine is an array of text section IDs, it might best to get the union of spine and text_sections.pluck(:id) and base the map on that... that way if a text section is removed, it will not longer show up in the spine. You can also use `find_by_id` instead of find, but then you could end up with blanks in the section map.